inspec/scripts/README.md

```diff
@@ -0,0 +1,318 @@
+
+
+
+ Use the AWS command line 
+
+Example 
+
+aws iam list-server-certificates
+
+
+List ebs volumes and status
+
+Using Python Boto3 Library
+
+import boto3
+ec2 = boto3.resource('ec2', region_name='us-west-2')
+volumes = ec2.volumes.all() # If you want to list out all volumes
+volumes = ec2.volumes.filter(Filters=[{'Name': 'status', 'Values': ['in-use']}]) # if you want to list out only attached volumes
+[volume for volume in volumes]
+
+
+List AWS Inventory
+
+Gather inventory of resources from AWS environment
+
+You need python 2 or 3
+pip install aws-list-all
+
+List all resources in an AWS account, all regions, all services(*). Writes JSON files for further processing.
+
+mkvirtualenv -p $(which python3) aws
+pip install aws-list-all
+aws-list-all query --region eu-west-1 --service ec2 --directory ./data/
+
+
+aws-list-all show data/ec2_*
+aws-list-all show --verbose data/ec2_DescribeSecurityGroups_eu-west-1.json
+
+
+Restricting the region and service is optional, a simple query without arguments lists everything. It uses a thread pool to parallelize queries and randomizes the order to avoid hitting one endpoint in close succession. One run takes around two minutes for me.
+
+
+
+Add immediate, more verbose output to a query with --verbose. Use twice for even more verbosity:
+
+aws-list-all query --region eu-west-1 --service ec2 --operation DescribeVpcs --directory data --verbose
+Show resources for all returned queries:
+
+aws-list-all show --verbose data/*
+Show resources for all ec2 returned queries:
+
+aws-list-all show --verbose data/ec2*
+List available services to query:
+
+aws-list-all introspect list-services
+List available operations for a given service, do:
+
+aws-list-all introspect list-operations --service ec2
+List all resources in sequence to avoid throttling:
+
+Get Cloudwatch Logs
+
+There is a python script that will pull json events for all of the streams from cloudwatch log and it will send the output to an S3 bucket.You
+need to provide a group_name, bucket name
+
+
+Get EBS volume and check to see if they are encrypted
+
+There is a python script the runs a report and uploads the report as a csv file into an s3 bucket of choice. The report list the volume, volume type, iops, size, created from a snapshot (if) the instances it is attached and if it is encrypted or not. The command takes argument of region if the instance has sufficient IAM role it should run. I know each environment is different and how you pass credentials will vary. You will need to go down the line in the code near the bottom where it says bucket and change to your bucket. You can also set the default regions in the code and dropp the argument --regions when executing the script
+
+Example:
+ebs-report.py --regions us-east-2 --file my-ebs-report.csv
+
+
+
+
+CloudTrail Logs
+
+Bash script to pull CloudTrail logs us just have to provide the values in the script Bucket Name Account Number and Region
+
+
+CloudTrail Lambda Function
```

Comment:
Is there automation for this, or are we not going to deploy this yet?  I would expect to see some terraform code ...

Reply:
I'm unsure if this is something that'll be deployed any time soon, this is something the security team might be able to help out with.